src/mlx/warnings/regex_checker.py

```diff
@@ -107,12 +161,156 @@ def check(self, content):
+                if isinstance((maximum := config[classification].get("max", 0)), (int, str)):
+                    checker.maximum = int(maximum)
+                if isinstance((minimum := config[classification].get("min", 0)), (int, str)):
+                    checker.minimum = int(minimum)
+                checker.cq_findings = self.cq_findings  # share object with sub-checkers
+                self.checkers[classification_lower] = checker
+            else:
+                print(f"WARNING: Unrecognized classification {key!r}")
+
+        for checker in self.checkers.values():
+            checker.cq_enabled = self.cq_enabled
+            checker.exclude_patterns = self.exclude_patterns
+            checker.cq_description_template = self.cq_description_template
+            checker.cq_default_path = self.cq_default_path
+
+
+class CoverityClassificationChecker(WarningsChecker):
+    SEVERITY_MAP = {
+        'false positive': 'info',
+        'intentional': 'info',
+        'bug': 'major',
+        'unclassified': 'major',
+        'pending': 'critical',
+    }
+
+    def __init__(self, classification, **kwargs):
+        """Initialize the CoverityClassificationChecker:
+
+        Args:
+            classification (str): The coverity classification
+        """
+        super().__init__(**kwargs)
+        self.classification = classification
+
+    @property
+    def cq_description_template(self):
+        ''' Template: string.Template instance based on the configured template string '''
+        return self._cq_description_template
+
+    @cq_description_template.setter
+    def cq_description_template(self, template_obj):
+        self._cq_description_template = template_obj
+
+    def return_count(self):
+        ''' Getter function for the amount of warnings found
+
+        Returns:
+            int: Number of warnings found
+        '''
+        return self.count
+
+    def add_code_quality_finding(self, match):
+        '''Add code quality finding
+
+        Args:
+            match (re.Match): The regex match
+        '''
+        finding = {
+            "severity": "major",
+            "location": {
+                "path": self.cq_default_path,
+                "positions": {
+                    "begin": {
+                        "line": 1,
+                        "column": 1
+                    }
+                }
+            }
+        }
+        groups = {name: result for name, result in match.groupdict().items() if result}
+        try:
+            description = self.cq_description_template.substitute(os.environ, **groups)
+        except KeyError as err:
+            raise WarningsConfigError(f"Failed to find environment variable from configuration value "
+                                      f"'cq_description_template': {err}") from err
+        if "classification" in groups:
+            finding["severity"] = self.SEVERITY_MAP[groups.get("classification", "unclassified").lower()]
```

Comment:
```suggestion
        if classification_raw := groups.get("classification"):
            finding["severity"] = self.SEVERITY_MAP[classification_raw.lower()]
```
Refactoring since `"major"` is already stored as the default severity level.